gsadjust/pyqt_models.py

```diff
@@ -1260,115 +1316,79 @@ def data(self, index, role=QtCore.Qt.DisplayRole):
                     m = index.model().itemFromIndex(index.sibling(index.row(), 0))
                 else:
                     m = index.model().itemFromIndex(index)
-                if type(m) is ObsTreeStation:
-                    if index.column() == 0:
-                        return m.display_name
-                    if index.column() == 1:
-                        if m.tmean != -999:
-                            return num2date(m.tmean).strftime('%Y-%m-%d %H:%M:%S')
-                        else:
-                            return m.tmean
-                    if index.column() == 2:
-                        return '{:.1f} ± {:.1f}'.format(m.gmean, m.stdev)
-                elif type(m) is ObsTreeLoop:
-                    if index.column() == 0:
-                        return m.name
-                    if index.column() == 1:
-                        return ''
-                    if index.column() == 2:
-                        return ''
-                elif type(m) is ObsTreeSurvey:
-                    if index.column() == 0:
-                        return m.name
-                    if index.column() == 1:
-                        return ''
-                    if index.column() == 2:
-                        return ''
+
+                return m.display_column_map.get(index.column(), '')
+
             elif role == QtCore.Qt.CheckStateRole:
                 if index.column() == 0:
                     m = index.model().itemFromIndex(index)
                     return m.checkState()
             elif role == QtCore.Qt.ToolTipRole:
-                if index.column() > 0:
-                    m = index.model().itemFromIndex(index.sibling(index.row(), 0))
-                else:
-                    m = index.model().itemFromIndex(index)
-                if type(m) is ObsTreeLoop or type(m) is ObsTreeSurvey:
-                    try:
-                        return m.tooltip
-                    except Exception as e:
-                        return ''
+                return m.tooltip
 
     def setData(self, index, value, role):
         if role == QtCore.Qt.CheckStateRole and index.column() == 0:
             m = index.model().itemFromIndex(index)
-            if value == QtCore.Qt.Checked:
-                m.setCheckState(QtCore.Qt.Checked)
-                if type(m) is ObsTreeLoop:
-                    self.signal_delta_update_required.emit()
-            else:
-                m.setCheckState(QtCore.Qt.Unchecked)
-                if type(m) is ObsTreeLoop:
-                    self.signal_delta_update_required.emit()
+            m.setCheckState(value)
```

Comment:
The value is in an `int` but so is `QtCore.Qt.Checked` etc. so we can apply the value directly.